python/pyarrow/types.pxi

```diff
@@ -2014,6 +2014,48 @@ cpdef KeyValueMetadata ensure_metadata(object meta, c_bool allow_none=False):
         return KeyValueMetadata(meta)
 
 
+cdef class FieldMergeOptions(_Weakrefable):
+    """
+    Options controlling how to merge the types of two fields.
+
+    By default, types must match exactly, except the null type can be
+    merged with any other type.
+
+    """
+
+    cdef:
+        CField.CMergeOptions c_options
+
+    __slots__ = ()
+
+    def __init__(self, *):
+        self.c_options = CField.CMergeOptions.Defaults()
```

Comment:
Do we want to expose all fine grained options in Python? (now it's only the "permissive" and "default")

While trying this out, I expected I could do something like `concat_tables(..., field_merge_options=pa.FieldMergeOptions(promote_integer_to_decimal=True)`

If we only allow the two main options, I am not sure it is worth actually exposing this class publicly. 

(exposing the fine grained options in python can also be a follow-up)

Reply:
I am certainly fine with not exposing the fine-grained options for this PR. But in that case we maybe also don't need this class? (and just accept the string version in the function that have this option)